### DISCUSSION TITLE
Why is imgui dear?

### DISCUSSION THREAD

**Discussion by MatijaBrown** [ocornut/imgui#4041] [Old/Archived discussions]
The title says it - why is it **Dear** ImGui? This is a bit confusing to me - could someone explain please?

**Comment by PathogenDavid** [comment]
Why is your name **Matija** Brown? 🤔 That's just what Dear ImGui is called!

Historically the library was just called "ImGui", but that's overly generic since there are other immediate mode GUI libraries besides Dear ImGui. (For example, [the Unity IMGUI module](https://docs.unity3d.com/ScriptReference/UnityEngine.IMGUIModule.html)) The rebranding to Dear ImGui happened back in 2015 to make the name more unique:

> [I] Would rather go for something friendly or an emotion than something very techy. The good thing about "Dear ImGui" is that it's sort of light-hearted, original and doesn't steal the show from the ImGui word.

You can read the full context and discussion in https://github.com/ocornut/imgui/issues/21

**Reply by MatijaBrown** [reply]
Oh - if it's meant that way that of course makes sense.
I thought the adjective _dear_ was purposefully chosen to describe Dear ImGui.
But this way... ...well, it's actually pretty obvious.
Thank you for answering the question! Very kind of you!

**Reply by igor98bn** [reply]
Your "the Unity IMGUI module" link doesn't work? Is the problem coming from me?

**Reply by MatijaBrown** [reply]
> Your "the Unity IMGUI module" link doesn't work? Is the problem coming from me?

No, I am the IMGUI.NET crashes with Silk.NET person.

**Reply by PathogenDavid** [reply]
@igor98bn The link still works for me. If you search `IMGUIModule` in the Unity scripting documentation it should come up.

**Comment by mrfrostbyte** [comment]
Thanks for the explanation. It was *sorely* needed. Now we understand the reason. But personally I still think it's . . . weird (to say the least) that _"dear"_ was chosen. It could've been a dozen other, more descriptive and sensible adjectives/words. Not that there's anything wrong with the word "dear", but most English speakers don't generally associate the word with software to start with.

Next, to quote @PathogenDavid, without the "dear" it would be _"overly generic"_, yet the adjective "dear", at least as it's used in most dialects (**and** most sociolinguistic registers of those dialects) of modern standard English, is still relatively generic semantically, and can take several semantic values, each of which may further be colored by surrounding context and other pragmatic functions. Yes, in general, it means "highly valued", "loved", "important", etc., which is all well and good. But those are only the most common meanings when used as an adjective, and the use of it as a modifier of a technical acronym like ImGui/IMGUI is uncommon and arguably pretty ambiguous.

But the primary reason I find it curious (to say the least) is because the word occupies a very important, very common and much different place in English semantics than merely as a common adjective describing something's value; namely, it is used as an opening when writing any informal or colloquial letter. Saying "Dear ImGui" brings to mind before anything else the misconception that someone is starting a letter addressed to this _ImGui_ "person"! 😜

All of these reasons, each of which alone could provide sufficient reason to select an alternate adjective or word for its purpose, when taken together cause a lot of confusion, and evidence of this is readily available simply by searching for the name _(Dear) ImGui_ (with or without the "Dear"). In fact, when I entered the simple search string "Why is it called Dear ImGui?" into three different common search engines, I found **literally** dozens of results similar to the question that began this comment (i.e., _"Why is imgui dear?"_). I am clearly not the only one who was confused by it lol!

However, despite all this, in summary I simply want to say that there's nothing inherently **wrong** with the word "dear" as used in this context. It's perfectly fine, uninsulting and maybe even makes sense to **some** people. The problem is, as we've seen, in its ambiguous and curious nature. That said, I mean no offense to the original developer of the "Dear ImGui" flavor of the library. I just thought I'd put in my two cents because (a) I have an extra 5 minutes and nothing to do, and (b) in the slight hopes that perhaps people will choose a more sensible word in similar cases, thus preventing a lot of confusion and rolling eyes. :wink:

Anyway, have a good one guys! Btw, the Dear ImGui incarnation of the IMGUI use case is very well implemented and I use it myself now that I know what it is. No offense to anyone. Just confused, like many others, clearly. 😎🖖🏼

**Reply by ocornut** [reply]
> most [...] don't generally associate the word with software to start with.

Seems good to me!

> The problem is, as we've seen, [...] its curious nature

Seems good to me!!

> Saying "Dear ImGui" brings to mind before anything else the misconception that someone is starting a letter 

Seems good to me!!!

I understand it's an unusual / weird / fun curious branding, I'm fine with that and wanted something not technical. 
Obviously it's not going to change anymore now. I think you are overthinking this, but arguably if its also designed to raise a eyebrow and make a people ask, I'm fine with that. I do also often asked how to pronounce it, and I have no answer for that haha!

**Reply by mrfrostbyte** [reply]
Thank you for sharing your opinion. I can't remember for sure, but I *think* I said it in my previous post, and even if I didn't, it bears saying, because it's a major part of my philosophy of life: I respect the opinions of others, even if I don't agree 100% with them. Basically I'm just saying I totally respect your opinions and views on the matter. I know the name of a certain software package is of the **smallest** importance in the grand scheme of things, and I think I realize that (imho) I just wasted a lot of time writing (and other peoples' time reading it) my lengthy response to the OP, I'm sure. But I just wanted to share *my* opinion as well, so there would be a balanced set of opinions on the conversation.

That said, although I still don't think the word "dear" is **ideal** in this context, there's no reason not to use it, and tbh, after reading your reply, I decided that I share your opinion for the most part. I hope I didn't cause any offense, as none was intended.

Finally, I'd like to give you my thanks again for replying and sharing, and for helping me understand better. I wish you good fortune and happiness, and hope you have a great day! 😄☮️

**Comment by NexusDevWorks** [comment]


**Reply by PathogenDavid** [reply]
Not sure where you heard that, Dear ImGui is specifically *not* a retained-mode UI framework.

**Reply by nicolasnoble** [reply]
Sounds like this was a ChatGPT answer...

**Reply by NexusDevWorks** [reply]
> Not sure where you heard that, Dear ImGui is specifically *not* a retained-mode UI framework.

I deleted my answer, but I'm still learning.

**Reply by NexusDevWorks** [reply]
> Sounds like this was a ChatGPT answer...

I'm guessing thats not the best place to learn?

**Reply by meshula** [reply]
ChatGPT is good for learning, its results can help you find true answers. Unfortunately, ChatGPT works by making things up, but pretends that it gives you facts. You have to check everything it tells you because it makes things up A LOT, but that checking process will teach you a lot as well because ChatGPT is at least smart about giving you key words and related concepts. If you just believe it though, it will cause you confusion again and again.
